### PR TITLE
Require PHP 5.6 and do a composer update

### PR DESCRIPTION
**Pull request by relaxnow**

### CODE REVIEW

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php [hunk 1]
```diff
@@ -62,10 +72,11 @@ public function authenticate(TokenInterface $token)
...
         $translatedAssertion = $this->attributeDictionary->translate($token->assertion);
 
-        $nameId      = $translatedAssertion->getNameID();
-        $institution = $translatedAssertion->getAttribute('schacHomeOrganization');
-        $email       = $translatedAssertion->getAttribute('mail');
-        $commonName  = $translatedAssertion->getAttribute('commonName');
+        $nameId         = $translatedAssertion->getNameID();
+        $institution    = $this->getInstitution($translatedAssertion);
+        $email          = $this->getEmail($translatedAssertion);
+        $commonName     = $this->getCommonName($translatedAssertion);
```
Comment (rjkip) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php]: This is why I voted for a helper method on `AssertionAdapter` in the SAML bundle, but oh well ¯\_(ツ)_/¯

Comment (DRvanR) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php]: here: `\`, seems you lost an arm :wink:

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php [hunk 2]
```diff
@@ -97,4 +108,96 @@ public function supports(TokenInterface $token)
...
+        $commonNames = $translatedAssertion->getAttributeValue('commonName');
+
+        if (empty($commonNames)) {
+            throw new BadCredentialsException(
+                'No commonName provided'
+            );
+        }
+
+        if (count($commonNames) > 1) {
+            $this->logger->notice('Multiple commonNames provided, picking first one', ['commonNames' => $commonNames]);
```
Comment (rjkip) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php]: 1. Is this behaviour accepted by SURF? If this is accepted, but action is required to prevent this, make sure that the IdP entity ID can be found somewhere in the paper trail (logs with the same request ID).
 2. We mustn't log PII (common name and e-mail). I think it would be wise to only mention the attribute name.

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php (hunk 1 above)
Comment (rjkip) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php]: I fix, yes?

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php (hunk 2 above)
Comment (relaxnow) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php]: > 1. Is this behaviour accepted by SURF?

I... I... I don't know, my behaviour has generally been pretty well received at SURF...
Heh, but good point, I'll ask @pmeulen 

> 2. We mustn't log PII (common name and e-mail). I think it would be wise to only mention the attribute name.

Ah yes, I heard about that, I'll remove it.

File: src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php [hunk 3]
```diff
@@ -97,4 +108,100 @@ public function supports(TokenInterface $token)
...
+        if (count($commonNames) > 1) {
+            throw new BadCredentialsException(
+                'Multiple commonName values provided in SAML Assertion'
+            );
+        }
+
+        $commonName = $commonNames[0];
+
+        if (!is_string($commonName)) {
+            $this->logger->warning('Received invalid commonName', ['commonNameCount' => count($commonName)]);
```
Comment (rjkip) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php]: This count seems irrelevant, but otherwise: :+1:

Comment (relaxnow) [inline src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php]: ```gettype``` is much better there 😄 .
Thanks!

**Comment by rjkip** [comment]
Let's a-go :star2:
